### PR TITLE
Add missing `FileLocationPatternConverter#requiresLocation()` (#2781)

### PR DESCRIPTION
**Pull request by dedeibel**
Fix for #2781

### CODE REVIEW

**Comment by ppkarwasz** [comment]
@dedeibel,

Nice catch. Could you also add a unit test that verifies that the pattern converters `%L`, `%l`, `%F`, `%C` and `%M` implement `LocationAware` and their `requiresLocation()` method returns `true`?

**Comment by vy** [comment]
@dedeibel, thanks so much for the fix. Would you mind porting the fix to the `2.x` branch with the changes @ppkarwasz requested in a separate PR, please?

**Comment by dedeibel** [comment]
> @dedeibel,
> 
> Nice catch. Could you also add a unit test that verifies that the pattern converters `%L`, `%l`, `%F`, `%C` and `%M` implement `LocationAware` and their `requiresLocation()` method returns `true`?

@ppkarwasz Thanks. I will try but please bear with me hat it might take some time to find my way around the tests.

@vy Yes I will. But please give me a some time.

**Comment by vy** [comment]
@dedeibel, no worries, take your time. Feel free to reach out to us.

Note that we'd prefer fixing `2.x` first, since Log4j 2 is heavily used and there is no Log4j 3 release date on the horizon.

File: log4j-core-test/src/test/java/org/apache/logging/log4j/core/pattern/LocationPatternConvertersRequireLocationTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.List;
+
+import org.apache.logging.log4j.core.LoggerContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+public class LocationPatternConvertersRequireLocationTest {
```
Comment (vy) [inline log4j-core-test/src/test/java/org/apache/logging/log4j/core/pattern/LocationPatternConvertersRequireLocationTest.java:28]:
```suggestion
class LocationPatternConvertersRequireLocationTest {
```

**Review by vy — APPROVED** [review]
@dedeibel, good job. :100:

@ppkarwasz, as in #2795, please go ahead and [squash] merge this PR, if you also agree with the changes.

**Pull request merged**